packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -491,7 +526,7 @@ export class Cluster extends ClusterBase {
       clusterSubnetGroupName: subnetGroup.clusterSubnetGroupName,
       vpcSecurityGroupIds: securityGroupIds,
       port: props.port,
-      clusterParameterGroupName: props.parameterGroup && props.parameterGroup.clusterParameterGroupName,
+      clusterParameterGroupName: Lazy.string({ produce: () => this.parseParameterGroup() }),
```

Comment:
I don't quite get why you want to use `Lazy` here.

Reply:
I thought `Lazy` had to be used if the parameter group could be created after the instantiation of the construct through the `addToParameterGroup` method

Reply:
pretty sure `Lazy` has to be used for post-synthesis setting of values, not post-constructor setting. There's plenty of cases of values being modified after the constructor terminates, without using `Lazy`s; see iam policies for a more dramatic example of this.